File: cpp/671_二叉树中第二小的节点_easy.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
private:
    int ans = -1;
public:
    int findSecondMinimumValue(TreeNode* root) {
        int rootVal = root->val;
        dfs(root->left, rootVal);
        dfs(root->right, rootVal);
        return ans;
    }
    void dfs(TreeNode* root, int rootVal) {
        if (!root)
            return;
        if (ans != -1 && root->val >= ans)
            return;
        if (root->val > rootVal)
            ans = root->val;
        dfs(root->left, rootVal);
        dfs(root->right, rootVal);
    }
};

File: python/671_二叉树中第二小的节点_easy.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def __init__(self):
        self.ans = -1
    
    def findSecondMinimumValue(self, root: TreeNode) -> int:
        rootVal = root.val
        def dfs(root):
            if not root:
                return
            if self.ans != -1 and root.val >= self.ans:
                return
            if root.val > rootVal:
                self.ans = root.val
            dfs(root.left)
            dfs(root.right)
        dfs(root.left)
        dfs(root.right)
        return self.ans
